exercise-1.js

```diff
@@ -294,234 +467,386 @@
+//your code...
+console.log(
+  "this line of code prints false because objects need to be stored in objects."
+);
 
-        //your code
+//loop through the mapObj and create a new array of only the data types, leaving out the example keys of the mapObj.  Use array methods to do this.  Example output : ['string',number','boolean',array','object']
+var objArr = [];
 
+var seq = mapObj.values();
+for (let i = 0; i < mapObj.size; i++) {
+  objArr.push(seq.next().value);
+}
 
-        // Once finished, declare a variable called futureValue that holds your stockGain function and enter any amount for the cost basis and a number for the number of years.  Run the function returned by the higher order function to display the future value of the stock.  
+console.log(objArr); // still working not done yety
+/************************************************************* */
+//Create 4 mathematical function expressions, add,subtract,multiply,divide.  put them in an array, and create a doMath() function that randomly selects one of the mathematical operations whenever it is invoked.  The doMath() function should print to the console the mathetmatical function that was carried out.  The doMath() function should return the computed value of any operation performed.
 
-        //your code...
+// ex : 2,3 => doMath(2,3) => adding : 5
 
+function doMath(x, y) {
+  let calc = Math.floor(Math.random() * operations.length);
+  console.log(operations[calc]);
+  return operations[calc](x, y);
+}
 
+//your code...
+const add = (x, y) => x + y;
+const subract = (x, y) => x - y;
+const multiply = (x, y) => x * y;
+const divide = (x, y) => x / y;
+
+const operations = [add, subract, multiply, divide];
+
+console.log(doMath(2, 3));
+/************************************************************* */
+//- Create a Higher Order Function called multiple(x) that takes a single parameter.  This HOF should return another function fn(y) that accepts another single parameter y.  This inner function should compute the product of it's parameter with the parameter passed into multiple.  Use this returned "first-class" function to compute triples of any given number.
+
+//your code...
+let hof = function multiple(x) {
+  return function fn(y) {
+    return x * y;
+  };
+};
+console.log(hof(10)(20));
+console.log(hof(hof(10)(20))(5)); //first value of hof gets entered into x, then gets multiplied by y
+
+//- Write an outer function called stockGain that has cost basis (basis) as a parameter; declare a variable called message that holds " is how much the stock has increased".  Return an inner function with years (yrs) as a parameter and declare a variable for growth rate (r) of 5%. Console log your calculation.
+
+//your code
+var stockGain = (basis) => {
```

Comment:
The expectation was for stockGain to return the inner function with 'yrs' as the argument.  That returned function would then be called using various 'yrs' values to print the required message.  The previous exercise would provide the logic necessary to accomplish this.